apps/devhub/utils.py

```diff
@@ -67,8 +67,14 @@ def limit_validation_results(validation):
     if lim and len(messages) > lim:
         # Sort messages by severity first so that the most important messages
         # are the one we keep.
-        TYPES = {'error': 0, 'warning': 1, 'notice': 2}
-        messages.sort(key=lambda msg: TYPES.get(msg.get('type')))
+        TYPES = {'error': 0, 'warning': 2, 'notice': 3}
+
+        def message_key(message):
+            if message.get('signing_severity'):
```

Comment:
I originally tested for presence but then I got concerned that at some point it will get set to `'signing_severity': None` and this would start treating everything the same. So I'm checking if it has a truthy value.